### DISCUSSION TITLE
Tracking Telemetry Across Multiple Vehicles

### DISCUSSION THREAD

**Discussion by jacben13** [nasa/openmct#7114] [Community Q&A]
Hello,
I have a use case where I am looking at multiple identical vehicles (2-10). I have implemented a telemetry/history server that receives telemetry updates from a python script that serves as a bridge. I would like the telemetry bridge to send telemetry for multiple vehicles, and be able to filter on OpenMCT for the specific vehicle I want to view telemetry for.

Is this a concept that's reasonable to implement? I could easily include a key/identifier with each telemetry update. Does anyone have an example of something like this?

Thanks!

**Comment by ozyx** [comment]
Hi @jacben13 !

Thanks for your interest in Open MCT. Have you checked out the [openmct-tutorial](https://github.com/nasa/openmct-tutorial) yet? That might provide a good starting point for what you're trying to accomplish. You might also have a look at [openmct-quickstart](https://github.com/scottbell/openmct-quickstart), which is a full-stack example project using Open MCT, CouchDB as persistence store, and YAMCS for telemetry (along with simulated telemetry data via a Python script).

Please let us know if this helps answer your question, or feel free to follow up here with any additional questions you may have.

Cheers,
Jesse

**Comment by jacben13** [comment]
Thanks for the response!

I should have been more specific, I followed the tutorial at https://github.com/nasa/openmct-tutorial to implement my initial application.

The application I currently have is able to display telemetry from a single vehicle, in order for me to see telemetry from another vehicle, I have to stop the python script, and restart it with a different argument to send a different vehicle's telemetry. What I think I need to be able to do is to have some form of filter on the OpenMCT front end, or a way to populate the object provider with an object of the same type for each vehicle.

The quickstart example is helpful - does it offer better flexibility in terms of multiple vehicles of the same type?

**Comment by akhenry** [comment]
@shefalijoshi

**Comment by shefalijoshi** [comment]
Hi @jacben13 

You can use Filters to narrow your views down to a specific vehicle. Your telemetry will need to include `filters` metadata for this to work. Filters are applied in the UI by navigating to the Object, Editing it and then selecting the Filters tab in the Inspector pane.

Documentation for implementing filters can be found here: https://github.com/nasa/openmct/blob/master/src/plugins/filters/README.md

For example: 
The `State Generator object type` included the tutorial has the following filter metadata:
```
filters: [
          {
            singleSelectionThreshold: true,
            comparator: 'equals',
            possibleValues: [
              { label: 'OFF', value: 0 },
              { label: 'ON', value: 1 }
            ]
          }
        ]
```
<img width="1377" height="448" alt="image" src="https://github.com/user-attachments/assets/f9a1076f-62f3-494c-9cee-e2a9801e9d50" />
